src/mgmt/srv.c

```diff
@@ -60,22 +61,87 @@ static struct daos_rpc_handler mgmt_handlers[] = {
 #undef X
 
 static void
-mgmt_drpc_handler(Drpc__Call *request, Drpc__Response **response)
+process_daos_request(Proto__DaosRank *rank, Proto__DaosResponse *response)
 {
-	Drpc__Response *resp;
+	/* response status is populated with SUCCESS on init */
+	D_DEBUG(DB_MGMT, "Received request to kill rank (%u) on pool (%s)\n",
+		rank->rank, rank->pool_uuid);
+
+	/* TODO: process request and populate daos response status */
+}
 
-	D_ALLOC_PTR(resp);
-	if (resp == NULL) {
-		D_ERROR("Failed to allocate response\n");
+static void
+mgmt_drpc_handler(Drpc__Call *request, Drpc__Response **response)
+{
+	Drpc__Response		*drpc_resp = NULL;
+	Proto__DaosRank		*daos_rank = NULL;
+	Proto__DaosResponse	*daos_resp = NULL;
+	uint8_t			*body = NULL;
+	size_t			len = 0;
+
+	D_ALLOC_PTR(drpc_resp);
+	if (drpc_resp == NULL) {
+		D_ERROR("Failed to allocate drpc response ref\n");
 		return;
 	}
 
-	drpc__response__init(resp);
-	resp->sequence = request->sequence;
-	resp->status = DRPC__STATUS__SUCCESS;
+	drpc__response__init(drpc_resp);
+	drpc_resp->sequence = request->sequence;
+	drpc_resp->status = DRPC__STATUS__SUCCESS;
+
+	switch (request->method) {
+	case DRPC_METHOD_MGMT_KILL_RANK:
+		/* Unpack the daos request from the drpc call body */
```

Comment:
there's enough logic in the kill rank method, seems like it makes sense to break it out into a helper function.